File: src/is_even.js
/**
 * Returns true if the number is even, false otherwise.
 * @param {number} num - The number to check.
 * @returns {boolean} - True if the number is even, false otherwise.
 * @throws {TypeError} - If the argument is not a number.
 * @throws {RangeError} - If the number is not an integer.
 * @throws {RangeError} - If the number is not a valid number.
 */
exports.isEven = function(num) {
    if (typeof num !== 'number') {
        throw new TypeError('The argument must be a number.');
    }
    if (!Number.isInteger(num)) {
        throw new RangeError('The number must be an integer.');
    }
    if (isNaN(num)) {
        throw new RangeError('The number must be a valid number.');
    }
    return num % 2 === 0;
}

File: src/is_even.zig
//! A module that provides a function to check if a number is even.
const std = @import("std");
const testing = std.testing;

/// Check if a number is even.
/// arg value: The number to check.
/// return: True if the number is even, false otherwise.
///
/// # Example
/// ```zig
/// const is_even = @import("is_even");
/// const std = @import("std");
/// const assert = std.testing;
/// assert.expect(is_even.isEven(2));
/// assert.expect(!is_even.isEven(3));
/// ```
export fn isEven(value: i32) bool {
    return @rem(value, 2) == 0;
}

test "is the number even" {
    try testing.expect(isEven(2));
    try testing.expect(!isEven(3));
    try testing.expect(isEven(48));
    try testing.expect(!isEven(49));
    try testing.expect(isEven(0));
    try testing.expect(isEven(-2));
    try testing.expect(!isEven(-3));
}
